src/parquet/util/input.cc

```diff
@@ -235,4 +234,49 @@ void InMemoryInputStream::Advance(int64_t num_bytes) {
   offset_ += num_bytes;
 }
 
+// ----------------------------------------------------------------------
+// ChunkedInMemoryInputStream
+ChunkedInMemoryInputStream::ChunkedInMemoryInputStream(RandomAccessSource* source,
+    MemoryAllocator* pool, int64_t start, int64_t end, int64_t chunk_size) :
+    source_(source), stream_offset_(start), stream_end_(end), buffer_offset_(0) {
+  buffer_ = std::make_shared<OwnedMutableBuffer>(chunk_size, pool);
+  chunk_size_ = buffer_->size();
+  buffer_offset_ = chunk_size_;
```

Comment:
Or you could wait to initialize buffer (since the buffer size is only a soft limit at the moment -- how does BufferedInputStream in Java handle this?)